shell/browser/serial/serial_chooser_controller.cc

```diff
@@ -235,5 +265,33 @@ void SerialChooserController::RunCallback(
     std::move(callback_).Run(std::move(port));
   }
 }
+void SerialChooserController::OnGetAdapter(
+    base::OnceClosure callback,
+    scoped_refptr<BluetoothAdapter> adapter) {
+  CHECK(adapter);
+  adapter_ = std::move(adapter);
+  adapter_observation_.Observe(adapter_.get());
+  std::move(callback).Run();
+}
+
+bool SerialChooserController::IsWirelessSerialPortOnly() {
+  if (allowed_bluetooth_service_class_ids_.empty()) {
+    return false;
+  }
+
+  // The system's wired and wireless serial ports can be shown if there is no
+  // filter.
+  if (filters_.empty()) {
+    return false;
+  }
+
+  // Check if all the filters are meant for serial port from Bluetooth device.
+  for (const auto& filter : filters_) {
+    if (!filter->bluetooth_service_class_id) {
+      return false;
+    }
+  }
```

Comment:
Asking to make sure I understand this paragraph:

`filters_` is a whitelist of tests, and Chromium sees all bluetooth as coming through wireless serial ports, so any filter with a bluetooth service class ID passes the "is this for wireless serial ports" question, is that correct?

Reply:
yes - correct!